src/agent/onefuzz-task/src/tasks/task_logger.rs

```diff
@@ -254,127 +263,204 @@ impl TaskLogger {
-        event: Receiver<LoggingEvent>,
-    ) -> Result<()> {
-        let initial_state = LoopContext {
-            log_writer,
-            pending_logs: vec![],
-            state: LoopState::Receive,
-            event,
-        };
-
-        let _loop_result = futures::stream::repeat(0)
-            .map(Ok)
-            .try_fold(initial_state, |context, _| async {
-                match context.state {
-                    LoopState::Send { start, count } => {
-                        match context
-                            .log_writer
-                            .write_logs(&context.pending_logs[start..start + count])
-                            .await?
-                        {
-                            WriteLogResponse::Success => {
-                                if start + count >= context.pending_logs.len() {
-                                    Result::<_, anyhow::Error>::Ok(LoopContext {
-                                        pending_logs: vec![],
-                                        state: LoopState::Receive,
-                                        ..context
-                                    })
-                                } else {
-                                    let new_start = start + 1;
-                                    let new_count = context.pending_logs.len() - new_start;
-                                    Result::<_, anyhow::Error>::Ok(LoopContext {
-                                        state: LoopState::Send {
-                                            start: new_start,
-                                            count: new_count,
-                                        },
-                                        ..context
-                                    })
-                                }
-                            }
-
-                            WriteLogResponse::MaxSizeReached => {
-                                Result::<_, anyhow::Error>::Ok(LoopContext {
-                                    state: LoopState::InitLog { start, count },
-                                    ..context
-                                })
-                            }
-                            WriteLogResponse::MessageTooLarge => {
-                                // split the logs here
+        context: LoopContext<T>,
+        flush_and_close: Option<(Instant, Duration)>,
+    ) -> Result<LoopContext<T>> {
+        match context.state {
+            LoopState::Send {
+                start,
+                count,
+                flush,
+            } => {
+                match context
+                    .log_writer
+                    .write_logs(&context.pending_logs[start..start + count])
+                    .await?
+                {
+                    WriteLogResponse::Success => {
+                        if start + count >= context.pending_logs.len() {
+                            if flush {
+                                bail!("done");
```

Comment:
Would `LoopState::Done` be more appropriate here? I know we break the loop for both Err and LoopState::Done:
https://github.com/microsoft/onefuzz/blob/853b8e99c599070479d87de31ca2f03f81f8280b/src/agent/onefuzz-task/src/tasks/task_logger.rs#L431-L443

But I think flushing the logs is a normal operation and not an error.

Reply:
good catch. This was supposed to be return the done state